### PR TITLE
Update asset and spec yaml for all vision models

### PR DESCRIPTION
**Pull request by gauravrajguru**

### CODE REVIEW

File: assets/models/system/stabilityai-stable-diffusion-2-1/model.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 path:
   container_name: finetuning-image-models
-  container_path: HuggingFace-mlflow/stabilityai-stable-diffusion-2-1/1702314884/mlflow_model_folder
+  container_path: HuggingFace-mlflow/stabilityai-stable-diffusion-2-1/1702902281/mlflow_model_folder
```
Comment (novaturient95) [inline assets/models/system/stabilityai-stable-diffusion-2-1/model.yaml:3]: hoping that this is expected in this PR?

File: assets/models/system/stabilityai-stable-diffusion-2-1/spec.yaml [hunk 2]
```diff
@@ -3,15 +3,36 @@ name: stabilityai-stable-diffusion-2-1
...
+      Standard_NC48ads_A100_v4,
+      Standard_NC4as_T4_v3,
+      Standard_NC64as_T4_v3,
+      Standard_NC8as_T4_v3,
+      Standard_NC96ads_A100_v4,
+      Standard_ND40rs_v2,
+      Standard_ND96amsr_A100_v4,
+      Standard_ND96asr_v4
+    ]
+version: 5
```
Comment (novaturient95) [inline assets/models/system/stabilityai-stable-diffusion-2-1/spec.yaml:38]: version was not incremented

**Review by novaturient95 — APPROVED** [review]

File: assets/models/system/stabilityai-stable-diffusion-2-1/model.yaml (hunk 1 above)
Comment (gauravrajguru) [reply assets/models/system/stabilityai-stable-diffusion-2-1/model.yaml:3]: Yes, the artifacts are changes for SD models. Taking this change in same pr.

File: assets/models/system/stabilityai-stable-diffusion-2-1/spec.yaml (hunk 2 above)
Comment (gauravrajguru) [reply assets/models/system/stabilityai-stable-diffusion-2-1/spec.yaml:33]: We have model ver 4 in prod registry. We don't need to update version.

**Review by novaturient95 — APPROVED** [review]

File: assets/models/system/automl-image-classification/asset.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 extra_config: model.yaml
 spec: spec.yaml
 type: model
-categories: ["Models"]
+categories: ["Foundation Models", "Automl"]
```
Comment (rjaincc) [inline assets/models/system/automl-image-classification/asset.yaml:4]:
```suggestion
categories: ["Foundation Models", "AutoML"]
```

Same at other places as well.

File: assets/models/system/compvis-stable-diffusion-v1-4/spec.yaml [hunk 4]
```diff
@@ -3,15 +3,31 @@ name: compvis-stable-diffusion-v1-4
 path: ./
 properties:
   SHA: 133a221b8aa7292a167afc5127cb63fb5005638b
-  datasets: LAION-5B
-  inference-min-sku-spec: 4|1|28|176
+  inference-min-sku-spec: 6|1|112|336
```
Comment (rjaincc) [inline assets/models/system/compvis-stable-diffusion-v1-4/spec.yaml:6]: why this change?

File: assets/models/system/facebook-deit-base-patch16-224/spec.yaml [hunk 5]
```diff
@@ -21,12 +26,98 @@ properties:
     Standard_NC12s_v3, Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3,
     Standard_NC24ads_A100_v4, Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4,
     Standard_ND96asr_v4, Standard_ND96amsr_A100_v4, Standard_ND40rs_v2
-  model_id: facebook/deit-base-patch16-224
 tags:
   Preview: ''
+  huggingface_model_id: facebook/deit-base-patch16-224
```
Comment (rjaincc) [inline assets/models/system/facebook-deit-base-patch16-224/spec.yaml:31]: No author for this one?

Comment (rjaincc) [reply assets/models/system/facebook-deit-base-patch16-224/spec.yaml:31]: Ignore - we have it later

File: assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/asset.yaml [hunk 6]
```diff
@@ -1,4 +1,4 @@
 extra_config: model.yaml
 spec: spec.yaml
 type: model
-categories: ["Models"]
+categories: ["Foundation Models", "open-mmlab"]
```
Comment (rjaincc) [inline assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/asset.yaml:4]:
```suggestion
categories: ["Foundation Models", "OpenMMLab"]
```

File: assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml [hunk 7]
```diff
@@ -21,12 +26,96 @@ properties:
     Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3, Standard_NC24ads_A100_v4,
     Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4,
     Standard_ND40rs_v2
-  model_id: mask_rcnn_swin-t-p4-w7_fpn_1x_coco
 tags:
   Preview: ''
+  SharedComputeCapacityEnabled: ''
```
Comment (rjaincc) [inline assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml:31]: What's the significance of this tag?

File: assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml [hunk 8]
```diff
@@ -21,12 +26,96 @@ properties:
...
+  open_mmlab_model_id: mask_rcnn_swin-t-p4-w7_fpn_1x_coco
```
Comment (rjaincc) [inline assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml:32]:
```suggestion
  openmmlab_model_id: mask_rcnn_swin-t-p4-w7_fpn_1x_coco
```

File: assets/models/system/mmd-3x-sparse-rcnn_r101_fpn_300-proposals_crop-ms-480-800-3x_coco/spec.yaml [hunk 9]
```diff
@@ -3,11 +3,16 @@ name: mmd-3x-sparse-rcnn_r101_fpn_300-proposals_crop-ms-480-800-3x_coco
...
 properties:
   SHA: fff646d3dda72d8c794471bfaa75b4db0adb7610
-  datasets: COCO
   evaluation-min-sku-spec: 4|1|28|176
-  evaluation-recommended-sku: Standard_NC6s_v3
+  evaluation-recommended-sku: Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3, Standard_NC12s_v3,
+    Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3,
+    Standard_NC24ads_A100_v4, Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4,
+    Standard_ND96asr_v4, Standard_ND96amsr_A100_v4, Standard_ND40rs_v2
   finetune-min-sku-spec: 4|1|28|176
```
Comment (rjaincc) [inline assets/models/system/mmd-3x-sparse-rcnn_r101_fpn_300-proposals_crop-ms-480-800-3x_coco/spec.yaml:11]: Should this be updated ?

File: assets/models/system/mmd-3x-vfnet_x101-64x4d-mdconv-c3-c5_fpn_ms-2x_coco/spec.yaml [hunk 10]
```diff
@@ -21,12 +26,96 @@ properties:
     Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3, Standard_NC24ads_A100_v4,
     Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4,
     Standard_ND40rs_v2
-  model_id: mmd-3x-vfnet_x101-64x4d-mdconv-c3-c5_fpn_ms-2x_coco
 tags:
   Preview: ''
+  SharedComputeCapacityEnabled: ''
+  open_mmlab_model_id: mmd-3x-vfnet_x101-64x4d-mdconv-c3-c5_fpn_ms-2x_coco
+  training_dataset: COCO
```
Comment (rjaincc) [inline assets/models/system/mmd-3x-vfnet_x101-64x4d-mdconv-c3-c5_fpn_ms-2x_coco/spec.yaml:33]: At some places we have "coco" and at other places we have "COCO". I think, better to align them.

Comment (gauravrajguru) [reply assets/models/system/mmd-3x-vfnet_x101-64x4d-mdconv-c3-c5_fpn_ms-2x_coco/spec.yaml:33]: done

File: assets/models/system/mmd-3x-sparse-rcnn_r101_fpn_300-proposals_crop-ms-480-800-3x_coco/spec.yaml (hunk 9 above)
Comment (gauravrajguru) [reply assets/models/system/mmd-3x-sparse-rcnn_r101_fpn_300-proposals_crop-ms-480-800-3x_coco/spec.yaml:11]: rearranged it in ascending order,

File: assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml (hunk 8 above)
Comment (gauravrajguru) [reply assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml:32]: done.

File: assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml (hunk 7 above)
Comment (gauravrajguru) [reply assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/spec.yaml:31]: Unclear, as per ayush - if workspace is out of quota then still it deployed on shared compute. Mandetory field to pass validation.

File: assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/asset.yaml (hunk 6 above)
Comment (gauravrajguru) [reply assets/models/system/mask_rcnn_swin-t-p4-w7_fpn_1x_coco/asset.yaml:4]: done

File: assets/models/system/automl-image-classification/asset.yaml (hunk 3 above)
Comment (gauravrajguru) [reply assets/models/system/automl-image-classification/asset.yaml:4]: done

File: assets/models/system/compvis-stable-diffusion-v1-4/spec.yaml (hunk 4 above)
Comment (gauravrajguru) [reply assets/models/system/compvis-stable-diffusion-v1-4/spec.yaml:6]: this was error in old spec, corrected it. Added min sku configuration of nc6.

**Review by novaturient95 — APPROVED** [review]

**Comment by novaturient95** [comment]
@gauravrajguru 

Please update PR description enumerating changes done in the PR. 
For one of the model where model_path was changed, please also add validation runs for the same
